### DISCUSSION TITLE
InlineCreate Operation -> error with the select field of type relationship

### DISCUSSION THREAD

**Discussion by ceeckhout** [Laravel-Backpack/community-forum#365] [Q&A (Help)]
I want to use a InlineCreate Operation next to the select field 'specification'. But when the InlineCreate Operation is implemented, the select field doesn't work anymore -> get 404 error page with the next message "The route admin/product/fetch/specification could not be found."

`CRUD::field('specification')
            ->type('relationship');
            ->inline_create(true);`

Anyone have an idea how I can solve this?

Thanks in advance!

**Comment by pxpm** [comment]
@ceeckhout hey there 👋 

Sorry for late reply. Hope you already figured out from the docs https://backpackforlaravel.com/docs/5.x/crud-fields#create-related-entries-in-a-modal-using-the-inlinecreate-operati

If you don't specify nothing in the `inline_create([])` configuration array, it will assume that you have setup the `FetchOperation` with a `public function fetchSpecification() { ... }` for it to use as a `data_source`. If using any other fetch name you can specify it with: `entity => ...` in the inline create configuration array. 

Alternativelly you can directly provide the `data_source => url(your-custom-url)`, just make sure the endpoint return the data in the format the field needs it. 

Let me know if I can help you with something else.

Cheers

**Reply by ceeckhout** [reply]
@pxpm Hi!

Thanks for your reply! I was absent for the past two weeks due to a medical operation.

If I understand correctly, you can use an InlineCreate Operation without having to use the Fetch Operation? If yes, how? Apparently the InlineCreate Operation wants to use the Fetch Operation, but I don't think I specified that anywhere.
Normally, there will be a maximum of 20 different specifications, which is why it is preferable not to call an AJAX call. 
Only one specification can be indicated for each product.

If I add the following; `->data_source(url("/admin/specification/inline/create"));` I no longer get a 404 error, mr it cannot load the specifications.

**Comment by cheadevit** [comment]
@pxpm Hello.

I tried to use inline create but I got some errors here is my code:

model:

```
    public function customer()
    {
        return $this->belongsTo(Customer::class, 'customer_id', 'id');
    }
```
```
public function fetchCustomer()
    {
        return $this->fetch(\App\Models\Customer::class);
    }

```
```
 $this->crud->addField([    // Relationship - everything is explicitly defined
            'label'         => 'Customer Name and Plate Number',
            'type'          => 'relationship',
            'name'          => 'customer',
            'attribute'     => 'name',
            'entity'        => 'customer',
            'model'         => "App\Models\Customer",
            'ajax'          => true,
            'minimum_input_length' => 0,
            'inline_create' => [
                'entity' => 'customer'
            ],
            'data_source' => backpack_url("checked-vehicle/fetch/customer"),
            'wrapperAttributes' => ['class' => 'form-group col-md-6'],
        ]);
 ```

**Reply by jcastroa87** [reply]
Hi @ceeckhout 

Did you add "FetchOperation" to your Controller?

```
use \Backpack\CRUD\app\Http\Controllers\Operations\FetchOperation;
```

What error you get?

Cheers.

**Comment by jcastroa87** [comment]
Due to not activity, I will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
